OPHD/Common.cpp

```diff
@@ -116,22 +116,46 @@ const std::map<IdleReason, std::string> IDLE_REASON_TABLE =
 };
 
 
-const std::array<std::string, 5> MoraleStringTable =
-{
-	constants::MoraleTerrible,
-	constants::MoralePoor,
-	constants::MoraleFair,
-	constants::MoraleGood,
-	constants::MoraleExcellent
+const std::array MoraleStringTable =
+{
+	std::string("Terrible"),
```

Comment:
I approve of cutting out the intermediate variables.

Reply:
Generally same here, I'd ultimately like to get rid of the Strings.h file and maybe use some sort of a global 'dictionary' that can load these values from a file at run-time.

I'm not a big fan of string literals in the code and prefer to either use named constants or a similar method to get them. It brings up the concern of non-latin language localization. Something that will be _really_ hard if not impossible with the current way things are handled (std::string really has no method of dealing with utf8 encoding and NAS2D's Font and text rendering don't handle it at all instead sticking with the ASCII definition of character sets... which could be fudged to localize but eh).

Anyway, yeah, the ultimate goal here for me is to move all of this from out of the `constants` namespace and into an object that does better translation using `class enum` values instead.

Reply:
A while back I was looking at the source code for [A Dark Room](https://github.com/doublespeakgames/adarkroom). I noticed they had an interesting translation mechanism. They had English strings in the code, but being looked up in a translation table. If nothing was loaded, the English would feed through to the output. Otherwise, a dictionary lookup would effectively translate each string and use that for the output. Looked like it made it pretty obvious to write the game for English speaking developers, while still allowing all game test to be translated with external translation files. The translation files paired up English phrases with the translated phrase.

Of course this sort of lookup is not context aware, so if you have the same English word used in two different ways in two different places, there would be no ability to translate it differently in those two places in other languages. Mind you, most strings were quite unique, so there was little risk of that happening in most places.

----

Luckily UTF-8 was designed to work well with most non-UTF-8 aware code. Aside from the `Font` rendering, it really shouldn't matter. It would also be possible to upgrade the `Font` rendering without disturbing any external code. The `Font` rendering interface is pretty clean in that regard. There would be no project wide updates needed to all calls to string rendering to make such an upgrade work. It would be very self contained.

Reply:
That's a neat idea but I think ultimately I'd rather go with a basic approach of using some sort of enumerated value that's used to lookup a string through a table. Simple, does the job without making too many assumptions, etc.

I'm more worried about the font rendering code than anything else? I'm not completely sure I know how to handle it but it would be a great feature moving forward.